packages/core/content-type-builder/admin/src/utils/getMaxDepth.ts

```diff
@@ -0,0 +1,79 @@
+import type { NestedComponent } from '../components/DataManagerProvider/utils/retrieveNestedComponents';
+import type { Internal } from '@strapi/types';
+
+type ChildComponent = {
+  repeatable: boolean;
+  component: Internal.UID.Component;
+  name?: string;
+  required?: boolean;
+  min?: number;
+};
```

Comment:
Why is a child component different to a regular component? This is the sort of context you should be writing about :)